backend/pkg/plugins/plugins_test.go

```diff
@@ -319,3 +319,42 @@ func TestPopulatePluginsCache(t *testing.T) {
 	require.True(t, ok)
 	require.Empty(t, pluginListArr)
 }
+
+// TestDelete checks the Delete function.
+func TestDelete(t *testing.T) {
+	tempDir, err := os.MkdirTemp("", "testdelete")
+	require.NoError(t, err)
+
+	defer os.RemoveAll(tempDir) // clean up
+
+	// Create a temporary file
+	tempFile, err := os.CreateTemp(tempDir, "testfile")
+	require.NoError(t, err)
+	tempFile.Close() // Close the file
+
+	// Test cases
+	tests := []struct {
+		pluginDir  string
+		pluginName string
+		expectErr  bool
+	}{
+		{pluginDir: tempDir, pluginName: tempFile.Name(), expectErr: false},          // Existing file
+		{pluginDir: tempDir, pluginName: "non-existent-directory", expectErr: false}, // Non-existent file
+		{pluginDir: tempDir, pluginName: "../", expectErr: true},                     // Directory traversal
```

Comment:
Maybe add another test where we can check for `isSubdirectory`?

Reply:
We do blackbox testing and since isSubdirectory is not exported it cannot be tested here. Also the test cases handle the sub directory case.